### PR TITLE
[mlir][vector] Remove unneeded mask restriction

### PR DESCRIPTION
**Pull request by jpienaar**
These were added when the only mapping was to LLVM.

### CODE REVIEW

File: mlir/include/mlir/Dialect/Vector/IR/VectorOps.td [hunk 1]
```diff
@@ -1819,17 +1819,17 @@ def Vector_MaskedLoadOp :
   Vector_Op<"maskedload">,
     Arguments<(ins Arg<AnyMemRef, "", [MemRead]>:$base,
                Variadic<Index>:$indices,
-               VectorOfRankAndType<[1], [I1]>:$mask,
-               VectorOfRank<[1]>:$pass_thru)>,
-    Results<(outs VectorOfRank<[1]>:$result)> {
+               VectorOf<[I1]>:$mask,
+               AnyVector:$pass_thru)>,
+    Results<(outs AnyVector:$result)> {
```
Comment (dcaballe) [inline mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:1824]: Probably good to add constraints to ensure that pass_thru and result's types match and that the number of elements in mask and one of the other vectors match as well.

File: mlir/include/mlir/Dialect/Vector/IR/VectorOps.td [hunk 2]
```diff
@@ -2140,14 +2140,14 @@ def Vector_CompressStoreOp :
   Vector_Op<"compressstore">,
     Arguments<(ins Arg<AnyMemRef, "", [MemWrite]>:$base,
                Variadic<Index>:$indices,
-               VectorOfRankAndType<[1], [I1]>:$mask,
-               VectorOfRank<[1]>:$valueToStore)> {
+               VectorOf<[I1]>:$mask,
+               AnyVector:$valueToStore)> {
```
Comment (dcaballe) [inline mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:2144]: For compress and expand, I would state in the doc that the compression/expansion applies only along the innermost dimension.

**Review by dcaballe — COMMENTED** [review]
Thanks, Jacques!

File: mlir/include/mlir/Dialect/Vector/IR/VectorOps.td (hunk 1 above)
Comment (jpienaar) [reply mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:1824]: Ha this let to finding a bug in the generated inference with variadics (well known limitations there)

File: mlir/include/mlir/Dialect/Vector/IR/VectorOps.td (hunk 2 above)
Comment (jpienaar) [reply mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:2144]: SG

**Pull request merged**